nightfall-optimist/src/services/block-assembler.mjs

```diff
@@ -168,25 +169,51 @@ export async function conditionalMakeBlock(proposer) {
           blockSize,
         });
 
-        // propose this block to the Shield contract here
+        // Propose this block to the State contract
         const txDataToSign = await stateContractInstance.methods
           .proposeBlock(
             Block.buildSolidityStruct(block),
             transactions.map(t => Transaction.buildSolidityStruct(t)),
           )
           .encodeABI();
-        const { address } = web3.eth.accounts.privateKeyToAccount(ethPrivateKey);
+
+        // Sign tx
         const blockStake = await stateContractInstance.methods.getBlockStake().call();
-        const tx = {
-          from: address,
-          to: stateContractInstance.options.address,
-          data: txDataToSign,
-          value: blockStake,
-          gas: 8000000,
-        };
-        const signedTx = await web3.eth.accounts.signTransaction(tx, ethPrivateKey);
-        const receipt = await web3.eth.sendSignedTransaction(signedTx.rawTransaction);
-        logger.debug(`Transaction receipt ${receipt}`);
+        const signedTx = await createSignedTransaction(
+          nonce,
+          ethPrivateKey,
+          ethAddress,
+          stateContractAddress,
+          txDataToSign,
+          blockStake,
+        );
+
+        // Submit tx and update db if tx is successful
+        // CHECK - db ops
+        txsQueue.push(async () => {
+          try {
+            const receipt = await sendSignedTransaction(signedTx);
+            logger.debug({ msg: 'Block proposed', receipt });
+
+            await removeTransactionsFromMemPool(block.transactionHashes);
```

Comment:
Maybe we could make use of `Promise.all([])` taking the these three calls inside it. e.g. 
```
await Promise.all([
  removeTransactionsFromMemPool(block.transactionHashes),
  removeCommitmentsFromMemPool(transactions.map(t => t.commitments.filter(c => c !== ZERO)).flat(Infinity)),
  removeNullifiersFromMemPool(transactions.map(t => t.nullifiers.filter(c => c !== ZERO)).flat(Infinity)),
]);
```
The code would be a bit cleaner and the behaviour it is the same.